src/flux/task-step.coffee

```diff
@@ -85,6 +85,14 @@ TaskStepConfig =
 
       step.content.questions?[0].formats?.indexOf('free-response') > -1
 
+    canTryAnother: (id) ->
+      step = @_get(id)
+      step? and
+        (step.has_recovery and step.correct_answer_id isnt step.answer_id) and
+        not Task.TaskStore.isTaskPastDue(step.task_id) and
```

Comment:
I'd like to but couldn't figure out a way to do so, since `isTaskPastDue` needs access to the task's `due_at`.  Regardless of whether the helper method was in a different location, the `task-step` would still have to access the task somehow to get that attribute.

I also thought about just changing the check in the component to be `not TaskStore.isTaskPastDue(..) and TaskStep. canTryAnother(...)`  but that felt like a lie since you can't try another if the task is past due.  The logic should be in the single method.

And then I thought about flipping the module loading hocus-pocus so the task store was the one that exported a blank object initially.  I rejected that because the TaskStep store calls only one method on Task, while Task calls two on TaskStep store.

So yeah... suggestions on how to break the circular dependency are most welcome :)